adapters/repos/db/vector/hnsw/startup.go

```diff
@@ -136,6 +136,10 @@ func (h *hnsw) restoreFromDisk() error {
 		}
 		h.cache.Drop()
 
+		if h.pqConfig.Segments == 0 {
+			h.pqConfig.Segments = int(state.PQData.Dimensions)
+		}
```

Comment:
Some comment would be useful: why segments can be == 0 and why it is changed to `dims`, not `dims/i`.